### PR TITLE
LAB- C110

### PR DESCRIPTION
**Pull request by Kgaur624**
added filter fetch tweet GET API
[LAB Task].   (https://confluence.dev.lithium.com/display/CD/Lab%3A+Java+streams)

### CODE REVIEW

File: projecty/src/main/java/com/service/Service.java [hunk 1]
```diff
@@ -71,11 +81,37 @@ public Status sendTweets(String args) throws NullPointerException, TwitterExcept
...
             logger.error("Error Occur", e);
-            //  throw new RuntimeException("Run time error");
+             throw new RuntimeException("Run time error");
         }
         return list;
     }
+
+    public List<TwitterData> filterTweet(String filter) throws IllegalArgumentException{
+        List<TwitterData> list= new ArrayList<>();
+        try {
```
Comment (vglalitha) [inline projecty/src/main/java/com/service/Service.java:93]: you can reduce the code ,by calling the getTimeline function.
Apply DRY.

File: projecty/test/testCases/FetchTweetsTest.java [hunk 2]
```diff
@@ -8,37 +8,57 @@
...
+        } catch (ParseException E) {
+            E.printStackTrace();
+        }
+        date = dateFormat.format(created);
+    }
 
     @Before
     public void setUp() {
         twitter= mock(Twitter.class);
-        //service= mock(Service.class);
```
Comment (vglalitha) [inline projecty/test/testCases/FetchTweetsTest.java:33]: remove the unwanted lines

File: projecty/src/main/java/com/service/Service.java (hunk 1 above)
Comment (Kgaur624) [reply projecty/src/main/java/com/service/Service.java:93]: hello @vglalitha 
as you told i have done these changes

File: projecty/src/main/java/com/resource/TweetController.java [hunk 3]
```diff
@@ -50,4 +47,15 @@ public Response sendTweet(PostReq postReq) throws TwitterException {
     public List<TwitterData> timeline() {
         return service.latestTweet();
     }
-}
\ No newline at end of file
+
+    @GET
+    @Path("/filter")
+    public List<TwitterData> filter(@QueryParam("value") String value) {
```
[the remark this answers is not in the document]
Comment (Kgaur624) [reply projecty/src/main/java/com/resource/TweetController.java:53]: hello @SupriyaPamulapati 
as you told I have done these changes

File: projecty/test/testCases/FetchTweetsTest.java (hunk 2 above)
Comment (Kgaur624) [reply projecty/test/testCases/FetchTweetsTest.java:33]: hello @vglalitha
as you told i have done these changes

**Review by SupriyaPamulapati — APPROVED** [review]

**Review by vglalitha — APPROVED** [review]

**Pull request merged**